### PR TITLE
Feature/orca 376 Standardize run_tests.sh and build_api.sh

### PR DESCRIPTION
**Pull request by AndrewDorn**
## Summary of Changes

Standardize run_tests.sh and build_api.sh

Addresses [ORCA-376: Standardize run_tests.sh and build_api.sh](https://bugs.earthdata.nasa.gov/browse/ORCA-376)

## Changes

* Created folder with initial version of build, test, and api scripts
* Migrated lambdas using old api method to new scripts

## Type of change

Workflow streamlining/standardization

## How Has This Been Tested?

Run locally.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the necessary documentation (remove those that do not apply)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] Unit tests
- [x] New and existing unit tests pass locally with my changes
    - [x] Automated tests passing (if not fork)
- [x] Any dependent changes have been merged and published in downstream modules
- [x] My code has passed security scanning
    - [x] Snyk
    - [x] git-secrets

### CODE REVIEW

File: tasks/orca_catalog_reporting/bin/run_tests.sh [hunk 1]
```diff
@@ -60,18 +60,54 @@ python3 -m venv venv
...
 check_rc $return_code "ERROR: pip install encountered an error."
 
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    orca_catalog_reporting.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/orca_catalog_reporting/bin/run_tests.sh:76]: `$return_code` not set. `let return_code=$?` line needs to occur before.

File: tasks/post_copy_request_to_queue/bin/run_tests.sh [hunk 2]
```diff
@@ -66,17 +66,47 @@ let return_code=$?
...
-if [ -d orca_shared ]; then
-    rm -rf orca_shared
-fi
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    post_copy_request_to_queue.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/post_copy_request_to_queue/bin/run_tests.sh:75]: `return_code` value is not being set prior to evaluation

File: tasks/post_to_catalog/bin/run_tests.sh [hunk 3]
```diff
@@ -66,11 +66,47 @@ let return_code=$?
 
 check_rc $return_code "ERROR: pip install encountered an error."
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    post_to_catalog.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/post_to_catalog/bin/run_tests.sh:75]: `return_code` is not set before the call.

File: tasks/post_to_database/bin/run_tests.sh [hunk 4]
```diff
@@ -66,11 +66,47 @@ let return_code=$?
 
 check_rc $return_code "ERROR: pip install encountered an error."
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    post_to_database.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/post_to_database/bin/run_tests.sh:75]: `return_code` is not set before the call.

File: tasks/request_status_for_granule/bin/run_tests.sh [hunk 5]
```diff
@@ -67,11 +67,47 @@ let return_code=$?
 check_rc $return_code "ERROR: pip install encountered an error."
 
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    request_status_for_granule.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/request_status_for_granule/bin/run_tests.sh:76]: `return_code` is not set before the call.

File: tasks/request_status_for_job/bin/run_tests.sh [hunk 6]
```diff
@@ -67,11 +67,47 @@ let return_code=$?
 check_rc $return_code "ERROR: pip install encountered an error."
 
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    request_status_for_job.py test
+check_rc $return_code "ERROR: Linting issues found."
```
Comment (bhazuka) [inline tasks/request_status_for_job/bin/run_tests.sh:76]: `return_code` is not set before the call.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Couple of minor items and a few thoughts in comments. I wonder if instead of the `lambda scripts` dir at the top level with a space if we shouldn't nest it and remove the space. Maybe `developer_tools/lambda_template` ? I am just thinking ahead as it may be a good place for us to store all our developer_tooling for environments, etc. as we start to get better and standardize a few things. maybe even a README to help. Thoughts?

File: lambda scripts/build.sh [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+## MAIN
+## -----------------------------------------------------------------------------
+## Create the build directory. Remove it if it exists.
+echo "INFO: Creating build directory ..."
+if [ -d build ]; then
+    rm -rf build
+fi
+
+mkdir build
+let return_code=$?
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/build.sh:58]: Replaced here and elsewhere

File: lambda scripts/build_api.sh [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+source venv/bin/activate
+
+## Install the requirements
+pip install -q --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org
+pip install -q "pydoc-markdown>=4.0.0,<5.0.0" --trusted-host pypi.org --trusted-host files.pythonhosted.org
+let return_code=$?
+
+check_rc $return_code "ERROR: pip install encountered an error."
+
+## Run the documentation command
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/build_api.sh:69]: Added here and elsewhere

File: lambda scripts/run_tests.sh [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+check_rc $return_code "ERROR: pip install encountered an error."
+
+
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    lambda_name.py test
+check_rc $return_code "ERROR: Linting issues found."
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:76]: Ah, yup, link check was just never applied. Should we put this in a separate PR? It's going to be messy.

File: lambda scripts/run_tests.sh [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+
+check_rc $return_code "ERROR: pip install encountered an error."
+
+
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    lambda_name.py test
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:75]: Seems to work. Changed here and elsewhere.

File: lambda scripts/run_tests.sh [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+
+echo "INFO: Sorting imports ..."
+isort \
+    --trailing-comma \
+    --ensure-newline-before-comments \
+    --line-length 88 \
+    --use-parentheses \
+    --force-grid-wrap 0 \
+    -m 3 \
+    lambda_name.py test
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:86]: Seems good. Switched here and elsewhere.

File: lambda scripts/run_tests.sh [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+echo "INFO: Formatting with black ..."
+black lambda_name.py test
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:89]: Seems good. Switched here and elsewhere.

File: lambda scripts/run_tests.sh [hunk 13]
```diff
@@ -0,0 +1,126 @@
...
+
+
+## Run code smell and security tests using bandit
+echo "INFO: Running code smell security tests ..."
+bandit -r lambda_name.py test
```
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:94]: Seems good. Switched here and elsewhere.

File: lambda scripts/run_tests.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (AndrewDorn) [reply lambda scripts/run_tests.sh:76]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/copy_to_glacier/README.md [hunk 14]
```diff
@@ -322,177 +322,4 @@ An example of a message is shown below:
...
-    FILE_HASH_TYPE_KEY = 'checksumType'
-    LOGGER = <cumulus_logger.CumulusLogger object>
-    List = typing.List
-    MB = 1048576
-    OS_ENVIRON_DEFAULT_STORAGE_CLASS_KEY = 'DEFAULT_STORAGE_CLASS'
-    OS_ENVIRON_ORCA_DEFAULT_BUCKET_KEY = 'ORCA_DEFAULT_BUCKET'
-    Union = typing.Union
-```
-
-## pydoc sqs_library.py
```
Comment (rizbihassan) [inline tasks/copy_to_glacier/README.md:461]: I think ## pydoc sqs_library.py should stay

**Review by rizbihassan — CHANGES_REQUESTED** [review]

File: tasks/copy_to_glacier/README.md (hunk 14 above)
Comment (AndrewDorn) [reply tasks/copy_to_glacier/README.md:461]: Added to api scripts here and elsewhere.

**Review by rizbihassan — APPROVED** [review]

**Review by bhazuka — APPROVED** [review]

File: tasks/copy_to_glacier/bin/run_tests.sh [hunk 15]
```diff
@@ -66,11 +66,47 @@ let return_code=$?
 check_rc $return_code "ERROR: pip install encountered an error."
 
 
+## Check code formatting and styling
+echo "INFO: Checking formatting and style of code ..."
+echo "INFO: Checking lint rules ..."
+flake8 \
+    --max-line-length 99 \
+    copy_to_glacier.py test
+check_rc $return_code "ERROR: Linting issues found."
```
[the remark this answers is not in the document]
Comment (bhazuka) [reply tasks/copy_to_glacier/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/orca_catalog_reporting/bin/run_tests.sh (hunk 1 above)
Comment (bhazuka) [reply tasks/orca_catalog_reporting/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/post_copy_request_to_queue/bin/run_tests.sh (hunk 2 above)
Comment (bhazuka) [reply tasks/post_copy_request_to_queue/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/post_to_catalog/bin/run_tests.sh (hunk 3 above)
Comment (bhazuka) [reply tasks/post_to_catalog/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/post_to_database/bin/run_tests.sh (hunk 4 above)
Comment (bhazuka) [reply tasks/post_to_database/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/request_status_for_granule/bin/run_tests.sh (hunk 5 above)
Comment (bhazuka) [reply tasks/request_status_for_granule/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

File: tasks/request_status_for_job/bin/run_tests.sh (hunk 6 above)
Comment (bhazuka) [reply tasks/request_status_for_job/bin/run_tests.sh]: https://bugs.earthdata.nasa.gov/browse/ORCA-505

**Pull request merged**